### ISSUE TITLE
Sometimes clicking on page fails to open it, need to click on another page then back to wanted page

### ISSUE THREAD

**Issue by hsandt** [zim-desktop-wiki/zim-desktop-wiki#2417]
**Describe the bug**
Unfortunately, I don't have a 100% repro, it just happens 1 time out of 20 or so.
Clicking on a page sometimes fail to open it. Maybe I did it too fast after a previous click. In any case, the old page is still displayed in the content pane.
Clicking a second time on the page does not force open it, because Zim doesn't register a page change.
I need to click on a different page, then back on the wanted page to force refresh and see the wanted page in the content pane.

**To Reproduce**
1. Click on some page, then another one, again and again, until you notice the content pane does not change
2. Verify that the content pane does not match selected page in the Index pane. This is the bug! 
3. Click again on the selected page: nothing changes
4. Click on a different page then back to the wanted page: this time, content pane will display it

**Expected behavior**
Clicking a page should always show its content on the Content pane

**Screenshots**
I forget to take a screenshot before doing step 4, but I'll take one next time it happens.

**Environment (please complete the following information):**
 - Version: 0.75.2
 - OS: Linux Ubuntu 22.04 with Unity desktop
 - Language: English

**Additional context**
I use a lot of Jump to, go back and forth with Alt+Left/Right and clicking on different pages. Maybe doing these increase the odds of reproducing the bug.

> For issues with no visible output / error, check the debug log for additional logging. This is available via the menu or by running "zim -D" from the commandline.

I see, I can try this next time when I launch Zim if I remember to do it.

**Comment by sojusnik** [comment]
Can confirm this, but also struggling to reproduce it.

Happens on Ubuntu 23.04 with Zim 0.75.1.

**Comment by hsandt** [comment]
I got it twice today!
Each time I was doing:
1. Create sub-page (should be auto-focused)
2. Click back on mother page
3. Sub-page is still focused

![image](https://github.com/zim-desktop-wiki/zim-desktop-wiki/assets/1067772/37b475fa-a4e0-4580-ae37-157a966cee53)

**Comment by sojusnik** [comment]
> I got it twice today! Each time I was doing:
> 
>     1. Create sub-page (should be auto-focused)
> 
>     2. Click back on mother page
> 
>     3. Sub-page is still focused

Good work! Same happens here as well.

**Comment by pfalcon** [comment]
You guys should start looking at the debug log one of these days, and post what you found in it. My bet is that there's some failing DB operation (of updating index, etc.). Which causes it.

**Comment by pfalcon** [comment]
And here's where the hunch comes from: https://github.com/zim-desktop-wiki/zim-desktop-wiki/issues/1285 . I was told that's the problem specific to may fork (https://github.com/zim-desktop-wiki/zim-desktop-wiki/issues/26), so d'oh.

**Comment by bruvellu** [comment]
Good catch, @hsandt! I've been experiencing this for a while, but could not figure out how to reproduce. It's still a bit tricky. For me, it only happens after creating a second sub-page:

[Screencast from 2023-07-15 18-06-03.webm](https://github.com/zim-desktop-wiki/zim-desktop-wiki/assets/361179/d7288c8a-6014-4c2c-a34e-d21e14852f6a)

After creating the first sub-page, I can click on the parent page. But after creating the second sub-page, I can't click on the parent. I have to click elsewhere beforehand.

And nothing shows up in the `--debug --verbose`, @pfalcon. Hard one to debug.

On Ubuntu 23.04, GNOME 44.2, Zim 0.75.2.

**Comment by marcocevoli** [comment]
+1 here. On my Win10 installation it happens when I want to return to the first (top) page coming from another page.
